### PR TITLE
LPD-6579 Applying and Configuring JS Client Extension with Script Element Attributes on a given page or page set

### PR DESCRIPTION
**Pull request by thiago-buarqque**
[LPD-18885](https://liferay.atlassian.net/browse/LPD-18885)

When configuring the JavaScript Client Extensions on a page (via Control Menu > Configure Page > Design > Customization > JavaScript):

If `async`/`defer` Boolean attributes are present in the CX, then:

- The value set in the CX will be used and the ability to change it when assigning to a page is be disabled.
    - When both attributes are set, `async` is going to be applied.
- If async/defer Boolean attributes are NOT PRESENT in the CX: the user can choose from the dropdown when applying on a page as-is now.
- If async/defer is set to `false`, the option will be removed from the dropdown.
- Both `data-senna-track` and `type` attributes should be able to be overridden.

## Preview

https://github.com/liferay-platform-experience/liferay-portal/assets/57292581/22962e75-7126-4b53-8250-23993410d03b

### CODE REVIEW

**Comment by thiago-buarqque** [comment]
ci:test:sf

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/layout/look_and_feel/GlobalJSCETsConfiguration.tsx [hunk 1]
```diff
@@ -377,6 +377,57 @@ function ExtensionRow({
 		},
 	];
 
+	const getScriptElementAttributesScope = () => {
```
Comment (ethib137) [inline modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/layout/look_and_feel/GlobalJSCETsConfiguration.tsx:380]: Hey @thiago-buarqque wouldn't these frontend changes need to be behind a feature flag? Is there a check somewhere else that is handling this?

Comment (thiago-buarqque) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/layout/look_and_feel/GlobalJSCETsConfiguration.tsx:380]: @ethib137 Oops, you are right. I should add the feature flag check [here](https://github.com/liferay-platform-experience/liferay-portal/pull/553/files#diff-cd84f38002311fafb756b32b7bdf1626b528626ffa0ed560d859ccba410780c8R54) and [here](https://github.com/liferay-platform-experience/liferay-portal/pull/553/files#diff-0e0293522d5c45f8ab182a328d4eb5aaeb917b406c3a965d31db9cb1ac91ba9aR446). I think checking on the backend is better than on the frontend in terms of performance.

Comment (thiago-buarqque) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/js/layout/look_and_feel/GlobalJSCETsConfiguration.tsx:380]: Done!

See [this](https://github.com/liferay-platform-experience/liferay-portal/pull/553/commits/310be97ac4972bd31896b745524eb152d3c3a8ff#diff-0e0293522d5c45f8ab182a328d4eb5aaeb917b406c3a965d31db9cb1ac91ba9aR450) and [this](https://github.com/liferay-platform-experience/liferay-portal/pull/553/commits/310be97ac4972bd31896b745524eb152d3c3a8ff#diff-cd84f38002311fafb756b32b7bdf1626b528626ffa0ed560d859ccba410780c8R58)

**Comment by thiago-buarqque** [comment]
ci:test:sf

**Comment by thiago-buarqque** [comment]
ci:test:relevant

**Comment by thiago-buarqque** [comment]
ci:reevaluate:1267401_13097

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutLookAndFeelDisplayContext.java [hunk 2]
```diff
@@ -439,6 +441,21 @@ private JSONObject _getCETJSONObject(
 			() -> typeSettingsUnicodeProperties.getProperty("loadType", null)
 		).put(
 			"name", cet.getName(_themeDisplay.getLocale())
+		).put(
+			"scriptElementAttributesJSON",
```
Comment (ethib137) [inline modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutLookAndFeelDisplayContext.java:445]: Hey @thiago-buarqque I've been playing around with the JS to try to make is simpler and I think the solution is actually moving the logic from the JS to here instead. 

Rather than pass all the `scriptElementAttributesJSON` to the frontend we can add `predefinedLoadType` and `loadTypeOptions`. We could calculate those values here and then the frontend logic will end up much simpler. What do you think?

Comment (ethib137) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutLookAndFeelDisplayContext.java:445]: Also if we were to keep the logic in the JS I refactored it [here](https://github.com/liferay/liferay-portal/commit/a47f973e79869edc6b26ea6c13fa5de817ae9cd2).

Comment (thiago-buarqque) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutLookAndFeelDisplayContext.java:445]: Hey @ethib137. 

We can't do it on the backend. The file you mentioned is called when the page loads. But when the client extension is chosen from the modal, it doesn't request anything else from the backend. The content that the frontend sees is the one passed from `getPayload` method inside [CETItemDescriptor](https://github.com/liferay-platform-experience/liferay-portal/blob/310be97ac4972bd31896b745524eb152d3c3a8ff/modules/apps/client-extension/client-extension-type-item-selector-web/src/main/java/com/liferay/client/extension/type/item/selector/web/internal/item/selector/CETItemDescriptor.java#L47). So if we were to build that on the BE, we would replicate the same logic on both `CETItemDescriptor` and `LayoutLookAndFeelDisplayContext`.

In summary, the FE looks the way to go. I'll apply the refactor you suggested tomorrow, I had to go home earlier today.

Comment (ethib137) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutLookAndFeelDisplayContext.java:445]: Okay... that makes sense. Thanks for looking into it. Be sure to test my changes to make sure I didn't break anything. Thanks.

Comment (thiago-buarqque) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutLookAndFeelDisplayContext.java:445]: @ethib137 just made the changes, things should be fine now. Thanks for the refactor on the frontend

**Review by ethib137 — APPROVED** [review]
LGTM. Feel free to forward when tests are passing.

**Comment by thiago-buarqque** [comment]
ci:forward

**Comment by thiago-buarqque** [comment]
Unrelated CI errors. Force forwarding to Brian

**Comment by thiago-buarqque** [comment]
ci:forward:force
